File: EEProjects/GhoBlog/backend/ghoblog/src/main/java/org/oiakushev/ghoblog/service/ProfileService.java
package org.oiakushev.ghoblog.service;

import org.oiakushev.ghoblog.dto.ProfileRequest;
import org.oiakushev.ghoblog.model.Profile;

public interface ProfileService {
    Profile getByEmail(String email);

    Profile add(ProfileRequest value);
}

File: EEProjects/GhoBlog/frontend/frontend/gho-blog/src/app/core/services/profile.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Profile } from '../models/profile.model';
import { ApiService } from './api.service';

@Injectable()
export class ProfileService {
    constructor (
        private apiService: ApiService
      ) {}

    getByEmail(email: string): Observable<Profile> {
        return this.apiService.get(`/private/api/profiles/email/${email}`);
    }

    add(item: any): Observable<Profile> {
        return this.apiService.post(`/private/api/profiles`, item);
    }
}